src/emulator/gdbstub/src/gdb.cpp

```diff
@@ -308,30 +371,76 @@ static std::string cmd_read_memory(HostState &state, PacketCommand &command) {
     return stream.str();
 }
 
-static std::string cmd_write_memory(HostState &state, PacketCommand &command) { return ""; }
+static std::string cmd_write_memory(HostState &state, PacketCommand &command) {
+    const std::string content = content_string(command);
+    const size_t pos_first = content.find(',');
+    const size_t pos_second = content.find(':');
+
+    const std::string first = content.substr(1, pos_first - 1);
+    const std::string second = content.substr(pos_first + 1, pos_second - pos_first);
+    const uint32_t address = parse_hex(first);
+    const uint32_t length = parse_hex(second);
+    const std::string hex_data = content.substr(pos_second + 1);
+
+    if (!check_memory_region(address, length, state.mem))
+        return "EAA";
+
+    for (uint32_t a = 0; a < length; a++) {
+        state.mem.memory[address + a] = static_cast<uint8_t>(parse_hex(hex_data.substr(a * 2, 2)));
+    }
+
+    return "OK";
+}
+
+// server_next() might not be able to tell the difference between the end of the packet ($) and 0x24 ($).
+// Thus, cmd_write_binary is disabled.
+static std::string cmd_write_binary(HostState &state, PacketCommand &command) {
+    const std::string content = content_string(command);
+    const size_t pos_first = content.find(',');
+    const size_t pos_second = content.find(':');
+
+    const
```

Comment:
if you have read crayon shinchan vol 23 page 15 then you will know that it's irrelevant to what I'm saying. This is some weird things

Reply:
pos_comma, pos_colon?

Reply:
I don't really understand but why the rest of the line is below?

Reply:
![image](https://user-images.githubusercontent.com/25717050/60614388-f9cd2e80-9df6-11e9-82b5-902246fbff76.png)

Reply:
I blame it on CLion.

Also I looked up Volume 23 of Crayon Shin-Chan and I am severely disappointed.
<img width="839" alt="Screen Shot 2019-07-03 at 2 00 04 PM" src="https://user-images.githubusercontent.com/32211852/60614533-10de3300-9d9b-11e9-84fd-447f2eb73f00.png">

Reply:
It wasn't popular on the East. I was just taking some random manga while my brain is high

![image](https://user-images.githubusercontent.com/25717050/60615028-64cb3500-9df8-11e9-8b72-1479046cc97c.png)

Here is the page though (in a language you may not know)